src/coreclr/jit/rangecheck.cpp

```diff
@@ -14,6 +14,11 @@
 //
 PhaseStatus Compiler::rangeCheckPhase()
 {
+    if (!doesMethodHaveBoundsChecks() || (fgSsaPassesCompleted == 0))
```

Comment:
just wondering, we have some phases where we do the check here in the `Phase`
and others where we do it before ever calling the phase

any particular reason for one over the other?

Reply:
I imagine that's just a stylistic inconsistency we've accumulated over time. I personally prefer this approach, since the phase name will still show up in JIT dumps, but we'll see it did nothing.

Reply:
It looks like in the outer scope we mostly check various configs and OptimizationEnabled, while various heuristics are used in the inner scope.